receiver/jaegerreceiver/config_test.go

```diff
@@ -110,3 +160,16 @@ func TestLoadConfig(t *testing.T) {
 			},
 		})
 }
+
+func TestFailedLoadConfig(t *testing.T) {
+	factories, err := config.ExampleComponents()
+	assert.Nil(t, err)
+
+	factory := &Factory{}
+	factories.Receivers[typeStr] = factory
+	_, err = config.LoadConfigFile(t, path.Join(".", "testdata", "bad_proto_config.yaml"), factories)
+	assert.NotNil(t, err)
```

Comment:
Would it be better to assert that `err` is a specific error from this list? https://github.com/open-telemetry/opentelemetry-collector/blob/60b03d0d2d503351501291b30836d2126487a741/config/config.go#L43

Should be likely `errUnmarshalErrorOnReceiver`.

Reply:
I was unable to assert the type b/c `errUnmarshalErrorOnReceiver` is private to that package.  I went with this.  If there's a better way, please let me know.
```
assert.EqualError(t, err, `error reading settings for receiver type "jaeger": must specify at least one protocol when using the Jaeger receiver`)
```
Should those err types be exported so that consuming packages can better check config failures?

Reply:
No, I think what you have is fine. No need to export err types.